packages/core/user-settings/core-user-settings-server-internal/user_settings_service.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { CoreContext } from '@kbn/core-base-server-internal';
+import { Logger } from '@kbn/logging';
+import { KibanaRequest } from '@kbn/core-http-server';
+
+/**
+ * @internal
+ */
+export interface InternalUserSettingsServiceSetup {
+  setUserProfileSettings: (client: UserProfileSettingsClientContract) => void;
+  getUserSettingDarkMode: (request: KibanaRequest) => Promise<string>;
+}
+
+export interface UserProfileSettingsClientContract {
+  get: (request: KibanaRequest) => Promise<Record<string, string>>;
+}
+
+export class UserSettingsService {
+  private logger: Logger;
+  private client?: UserProfileSettingsClientContract;
+
+  constructor(coreContext: CoreContext) {
+    this.logger = coreContext.logger.get('user-settings-service');
+  }
+
+  public setup(): InternalUserSettingsServiceSetup {
```

Comment:
Question: Do Services require a `start()` and `stop()`

Reply:
Not strictly no. In theory access methods such as `getUserSettingDarkMode` should be exposed from the `start` contract, but given how ugly/hacky the `RenderingService` parameters works atm, I'll accept having it exposed from `setup` given it's only internal.